document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/SmallFullAccountsDataHandler.java

```diff
@@ -73,30 +81,38 @@ public DocumentInfoResponse getSmallFullAccountsData(Transaction transaction, St
         }
     }
 
-    private CompanyAccounts getCompanyAccounts(String resourceLink, String requestId) throws HandlerException {
-
-        CompanyAccounts accounts;
+    private Transaction getTransaction(CompanyAccounts companyAccounts, String requestId, String resourceId)
+            throws HandlerException {
 
+        String transactionId = getTransactionId(companyAccounts, requestId, resourceId);
```

Comment:
This could fail early (throw an exception if the transaction ID is null without calling the transactions API).

Reply:
agree I will pull this out of the previous method and make a check to see if it is null, if it is I'll throw an exception.

Reply:
updated